File: JAVASCRIPT/03-FUNCIONES/practica-palindromo/palindromo.js
const palindromo = (palabra) => {

    palabra = palabra.replace(' ','');
    palabra = palabra.toLowerCase();

    let palabraInvertida = palabra.split('').reverse().join('');

    // console.log(palabrainvertida);

    if (palabra == palabraInvertida) {
        return true;
    } else {
        return false;
    }
}

// console.log(palindromo('Luz azul'));

let palabra = prompt('Ingrese una palabra');

let esPalindromo = palindromo(palabra);

if (esPalindromo) {
    document.write(`
    <h1>
    ${palabra} es un palíndromo
    </h1>`);
} else {
    document.write(`<h1>
    ${palabra} NO es un palíndromo
    </h1>`);
}

File: GO/src/golang/04-strings-strconv/01-palindromo/palindromo.go
package main

import (
	"fmt"
	"strings"
)

func reverse(cadena string) string {
	arrayCadena := strings.Split(cadena, "")
	arrayInvertida := make([]string, 0)

	for i := len(arrayCadena) - 1; i >= 0; i-- {
		arrayInvertida = append(arrayInvertida, arrayCadena[i])
	}
	return strings.Join(arrayInvertida, "")
}

func esPalindromo(palabra string) bool {
	fmt.Println(palabra)

	palabra = strings.ToLower(palabra)
	palabra = strings.ReplaceAll(palabra, " ", "")

	palabraInvertida := reverse(palabra)
	return palabra == palabraInvertida
}

func main() {
	// Palabras palíndromas: se leen igual de derecha a izquierda y vicerversa.

	if esPalindromo("ana") {
		fmt.Println("es palíndromo")
	} else {
		fmt.Println("NO es palíndromo")
	}
}

// Salida: es palíndromo
